ra_aid/agent_utils.py

```
@@ -56,58 +61,189 @@
+    first_tokens = estimate_messages_tokens([first_message])
+    new_max_tokens = max_tokens - first_tokens
+
+    trimmed_remaining = trim_messages(
+        remaining_messages,
+        token_counter=estimate_messages_tokens,
+        max_tokens=new_max_tokens,
+        strategy="last",
+        allow_partial=False,
+    )
+
+    return [first_message] + trimmed_remaining
```

Ruff (0.8.2)

Consider `[first_message, *trimmed_remaining]` instead of concatenation

Replace with `[first_message, *trimmed_remaining]`

(RUF005)